app/usecases/tasks.js

```diff
@@ -1,41 +1,86 @@
-const msTodo = require("../services/todo")();
 
-module.exports = function() {
+module.exports = function(db, oAuth2Client) {
+  const msTodo = require("../services/todo")(db);
   const preferences = require("../services/preferences")(db);
+  const cal = require("../services/gcalendar")(db, oAuth2Client);
+  const gsearch = require("../services/gsearch");
 
   this.onUpdate = (ctx, waRes)=>{
     if (waRes.generic[0].text === "tasks_todo_auth") {
       msTodo.authorizeUser(ctx);
     } else if (waRes.generic[0].text === "tasks_show") {
       msTodo.getTodos().then((todos)=>{
+        let today = new Date();
+        today.setHours(9);
+        today.setMinutes(0);
+        today.setSeconds(0);
+        if (today.getTime() <= new Date().getTime()) {
+          today = new Date();
+        }
+        const tomorrow = new Date(today);
+        tomorrow.setDate(tomorrow.getDate() + 1);
         // show tasks to user
         let inlineKeyboardMarkup = {inline_keyboard: [[]]};
         let toSend = "Du hast folgende Aufgaben offen:\n";
         if (todos.length != 0) {
           todos.forEach((todo, i)=>{
             toSend += `- ${todo.Subject}\n`;
           });
-          toSend += "\n" + `Ich empfehle dir, dass du mit ${todos[0].Subject} beginnst, es ist am frühsten fällig :)\n`;
-          toSend += "Gib mir Bescheid, wenn du die Aufgabe erledigt hast oder abbrichst:";
 
-          inlineKeyboardMarkup = {inline_keyboard: [[{
-            text: "Erledigt",
-            callback_data: "tasks_exercise_done",
-          }], [{
-            text: "Braucht noch Zeit",
-            callback_data: "tasks_exercise_notdone",
-          }]]};
+          cal.getBusySlotsByCalendarId(today.toISOString(), tomorrow.toISOString(), "primary").then((calRes)=>{
```

Comment:
FYI: Alle Funktionen von Moment.js sind mutable, heißt, wenn du `today`als Variable in ihrem Ausgangszustand behalten möchtest, solltest du `moment(today).toISOString()` machen. Das ist in diesem Fall nicht so tragisch, aber eventuell, falls du `date.add(...)` oder so nutzt.

Reply:
ich benutze garkein Moment, ich mag die Date objekte von JavaScript ganz gerne :)

Reply:
Wir müssen schauen, ob das gut ist, dass wir Datum nicht einheitlich verwenden... aber in deinem Fall dachte ich einfach, dass `.toISOString()`eine Moment-spezifische Funktion ist. Ich wusste nicht, dass das auch beim nativen `Date()`-Objekt geht. :D

Reply:
Ja ich hatte wie du siehst bis jetzt alles ohne Probleme mit dem Date Objekt machen können und keinen need gesehen Moment zu benutzen :) für richtige, größere Projekte wäre das sicherlich nützlich, wenn man sich auf vanilla-Date oder Moment-Date einigt. Resolve den comment einfach wenn das damit erledigt wäre.